features/polyExplorer/src/components/clusterStories/receivingCompanies.jsx

```diff
@@ -1,74 +1,141 @@
 import React, { useContext, useMemo } from "react";
-import { Tabs, Tab } from "@polypoly-eu/poly-look";
+import { Tabs, Tab, PolyChart } from "@polypoly-eu/poly-look";
 
 import i18n from "../../i18n.js";
 import { ExplorerContext } from "../../context/explorer-context.jsx";
 
-const Companies = ({ entities }) => (
-    <table>
-        <tbody>
-            {entities.map(({ name, dataRecipients }, index) => (
-                <tr key={index}>
-                    <td>{name}</td>
-                    <td>{dataRecipients.length}</td>
-                </tr>
-            ))}
-        </tbody>
-    </table>
-);
+import "./receivingCompanies.css";
 
-function Industries({ entities }) {
-    const { companies } = useContext(ExplorerContext);
-    const recipients = [].concat.apply(
-        [],
-        entities.map(({ dataRecipients }) => dataRecipients)
+const chartColors = {
+    primary: "#7ee8a2",
+    secondary: "#0f1938",
+};
+
+class IndexedLegend {
+    constructor(items) {
+        this.items = items;
+    }
+
+    labelOf(item) {
+        const index = this.items.indexOf(item);
+        if (index === -1) return "";
+        return `${index + 1}`;
+    }
+
+    render() {
+        return (
+            <div className="indexed-legend">
+                {this.items.map((item, index) => (
+                    <div key={index}>
+                        {this.labelOf(item)}: {item}
+                    </div>
+                ))}
+            </div>
+        );
+    }
+}
+
+function Companies({ entities }) {
+    const legend = new IndexedLegend(entities.map(({ name }) => name));
+    const data = entities.map(({ name, dataRecipients }) => ({
+        title: legend.labelOf(name),
+        value: dataRecipients.length,
+    }));
+    return (
+        <>
+            <PolyChart
+                type="vertical-bar-chart"
+                data={data}
+                barColor={chartColors.primary}
+                barValueColor={chartColors.secondary}
+            />
+            {legend.render()}
+        </>
     );
+}
+
+function extractRecipientsPerIndustry(entities, companies) {
+    const sharingCounts = {};
+    for (let { dataRecipients } of entities)
+        for (let recipient of dataRecipients)
+            sharingCounts[recipient] = (sharingCounts[recipient] || 0) + 1;
 
-    const companiesPerIndustry = useMemo(() => {
-        const map = {};
-        for (let recipient of recipients) {
-            const industry = companies[recipient]?.industryCategoryName();
-            if (!industry) {
-                console.error(`Unable to determine industry of ${recipient}`);
-                continue;
-            }
-            map[industry] = (map[industry] || 0) + 1;
+    const result = {};
+    for (let [recipient, sharingCount] of Object.entries(sharingCounts)) {
+        const industry = companies[recipient]?.industryCategoryName();
+        if (!industry) {
+            console.error(`Unable to determine industry of ${recipient}`);
```

Comment:
This was in the original code, but are we going to keep it? Or maybe raise an exception?

Reply:
Also my changes, I don't think @mteresa-jimenez touched this code at all. I wanted to leave an error so we remember we have some issues in the data, but I didn't want the whole thing to break either.